src/main/java/seedu/address/model/AddressBook.java

```diff
@@ -15,6 +16,7 @@
 public class AddressBook implements ReadOnlyAddressBook {
 
     private final UniquePersonList persons;
+    private Pair<Integer, Integer> indexPairOfOngoingVisit = new Pair<>(-1, -1);
```

Comment:
I am not sure if UI details should be tracked in `AddressBook`. In my view `AddressBook` should only be responsible for the patient data itself. Introducing UI metadata into the class may cause issues with undo/redo/history as this means that navigation commands like `view-visit` etc. would alter the `AddressBook`, meaning they would have to be `MutableCommand`, meaning they would be included in the undo history. This would go against the purpose of undo/redo as it is meant to affect the latest data change; it would be troublesome to have to undo all the UI-navigating commands to reach the last data-modifying command. There are also performance/memory concerns as a copy of the entire `AddressBook` is saved after each `MutableCommand`.

Perhaps such UI metadata could be stored in `preferences.json`? The file already stores things like window size. Even if the `indexPairofOngoingVisit` somehow goes out of sync with the actual `AddressBook` data, the consequences would be minor as it is only needed for convenience when the app starts up. Even if the pair is invalid, the app could simply start from the default view.

Reply:
Hi Gabriel, thanks for your well thought out review. I thought about it abit more and I think it's better to keep it as it is. 

To clarify the purpose of ```indexPairOfOngoingVisit```, this variable is to track the ```Person``` and his ```Visit``` that is currently being marked as the ongoing `Visit`. In the current use case, the user must execute `visit-start p/PATIENT_INDEX a/APPOINTMENT_INDEX` to convert an appointment into a `Visit` and only then is that `Visit` considered an ongoing `Visit` until the user marks that `Visit` as done using `visit-finish`. It's not a UI metadata as there can only be one ongoing `Visit` at any point in time (since the nurse can only visit one `Person` at any time), and this ongoing `Visit` can only be finished using the `visit-finish` command (regardless of the execution of other commands like `visit-view`)

Upon creating a visit using `visit-start`, the `Patient` will automatically be allocated the `Visit`. However, how do you keep track of which `Patient` and which `Visit`? This is the purpose of `indexPairOfOngoingVisit` which keeps track of the index of the ongoing `Visit`.

In between `visit-start` and `visit-finish`, the user can do various actions towards the ongoing visit such as update the `Visit Tasks`' details, mark them as done, modify its remarks, display the `Visit` or finish the `Visit`. These changes will automatically be applied to the `Visit` itself.

In the event that the application terminates before the `Visit` can be finished, the `Visit` data will still be intact, but the reference to which `Visit` is the ongoing `Visit` will be lost. This is why `indexPairOfOngoingVisit` is stored in the `AddressBook` to ensure that after the app restarts, the linkage still exists.

The rationale behind putting it into `AddressBook` is because `AddressBook` is currently _the_ class that stores the data of the application and I felt that since `indexPairOfOngoingVisit` was effectively a pointer to said data, its place in `AddressBook` (which will most probably be renamed in the future) was sufficiently appropriate. The alternative to this is to create a `VisitMetaData` class, which will bubble changes in `UserPrefs`, `ReadonlyAddressBook`, `Storage` etc just to include this pair of integers which will still have to refer to the data provided by `AddressBook`. The tradeoff from the abstraction appears nominal.

Reply:
I see. So `indexPairOfOngoingVisit` prevents other visits from being started while there is a current one ongoing, while still allowing the nurse to `visit-view` other visits in read-only mode if needed. This does seem like a good rationale. Approved.

On a more tangential note, could `visit-start` be simplified to `visit-start p/PATIENT_INDEX [a/APPOINTMENT_INDEX]` instead of `visit-start p/PATIENT_INDEX a/APPOINTMENT_INDEX`  (i.e. make the appointment linkage optional)? Let's say the nurse wants to start a visit on-the-fly due to an emergency call or similar. In such a case there would be no appointment scheduled in advance. Making `a/` optional would mean the nurse wouldn't have to create an appointment first before starting the visit.

Reply:
That is a good idea. I will think more on it and how it affects appointments.